src/Model/UserItem.php

```diff
@@ -0,0 +1,281 @@
+<?php
+
+/**
+ * @file src/Model/UserItem.php
+ */
+
+namespace Friendica\Model;
+
+use Friendica\Core\Logger;
+use Friendica\Core\Hook;
+use Friendica\Database\DBA;
+use Friendica\DI;
+use Friendica\Util\Strings;
+
+class UserItem
+{
+	// Notification types
+	const NOTIF_NONE = 0;
+	const NOTIF_EXPLICIT_TAGGED = 1;
+	const NOTIF_IMPLICIT_TAGGED = 2;
+	const NOTIF_THREAD_COMMENT = 4;
+	const NOTIF_DIRECT_COMMENT = 8;
+	const NOTIF_COMMENT_PARTICIPATION = 16;
+	const NOTIF_ACTIVITY_PARTICIPATION = 32;
+	const NOTIF_SHARED = 128;
+
+	/**
+	 * Checks an item for notifications and sets the "notification-type" field
+	 *
+	 * @param int $iid Item ID
+	 */
+	public static function setNotification(int $iid)
+	{
+		$fields = ['id', 'uid', 'body', 'parent', 'gravity', 'tag', 'contact-id', 'thr-parent', 'author-id'];
+		$item = Item::selectFirst($fields, ['id' => $iid, 'origin' => false]);
+		if (!DBA::isResult($item)) {
+			return;
+		}
+
+		// fetch all users in the thread
+		$users = DBA::p("SELECT DISTINCT(`contact`.`uid`) FROM `item`
+			INNER JOIN `contact` ON `contact`.`id` = `item`.`contact-id` AND `contact`.`uid` != 0
+			WHERE `parent` IN (SELECT `parent` FROM `item` WHERE `id`=?)", $iid);
+		while ($user = DBA::fetch($users)) {
+			self::setNotificationForUser($item, $user['uid']);
+		}
+		DBA::close($users);
+	}
+
+	/**
+	 * Checks an item for notifications for the given user and sets the "notification-type" field
+	 *
+	 * @param array $item Item array
+	 * @param int   $uid  User ID
+	 */
+	private static function setNotificationForUser(array $item, int $uid)
+	{
+		$thread = Item::selectFirstThreadForUser($uid, ['ignored'], ['iid' => $item['parent'], 'deleted' => false]);
+		if ($thread['ignored']) {
+			return;
+		}
+
+		$notification_type = self::NOTIF_NONE;
+
+		if (self::checkShared($item, $uid)) {
+			$notification_type = $notification_type | self::NOTIF_SHARED;
+		}
+
+		$profiles = self::getProfileForUser($uid);
+
+		// Fetch all contacts for the given profiles
+		$contacts = [];
```

Comment:
What is the functional difference between `$profiles` and `$contacts`?

Reply:
One is containing the contact IDs, the other one the profile links.